lib/sqlitex/sql_builder.ex

```diff
@@ -0,0 +1,74 @@
+defmodule Sqlitex.SqlBuilder do
+  @moduledoc """
+  This module contains functions for SQL creation. At the moment
+  it is only used for `CREATE TABLE` statements.
+  """
+  
+  # Supported table options
+  defp table_opt(:temporary), do: {:temp, "TEMP"}
+  defp table_opt(:temp), do: {:temp, "TEMP"}
+
+  defp table_opt({:primary_key, cols}) when is_list(cols) do
+    {:primary_key, ",PRIMARY KEY (" <> Enum.join(cols, ",") <> ")"}
+  end
+
+  defp table_opt({:primary_key, col}) when is_atom(col) do
+    {:primary_key, ",PRIMARY KEY (" <> Atom.to_string(col) <> ")"}
+  end
+
+  # Catch invalid / unsupported table options
+  defp table_opt(opt), do: raise "Unkown table option: #{IO.inspect(opt)}"
```

Comment:
You don't need to raise an error here.  Unmatched arguments will raise runtime exceptions anyway.